astropy/convolution/boundary_extend.pyx

```diff
@@ -38,56 +40,36 @@ def convolve1d_boundary_extend(np.ndarray[DTYPE_t, ndim=1] f,
 
     # release the GIL
     with nogil:
-        # Need a first pass to replace NaN values with value convolved from
-        # neighboring values
         for i in range(nx):
-            if npy_isnan(f[i]):
-                top = 0.
-                bot = 0.
-                iimin = i - wkx
-                iimax = i + wkx + 1
-                for ii in range(iimin, iimax):
-                    iii = int_min(int_max(ii, 0), nx - 1)
-                    val = f[iii]
-                    if not npy_isnan(val):
-                        ker = g[<unsigned int>(wkx + ii - i)]
-                        top += val * ker
-                        bot += ker
-
-                if bot != 0.:
-                    fixed[i] = top / bot
-                else:
-                    fixed[i] = f[i]
-            else:
-                fixed[i] = f[i]
+            fixed[i] = f[i]
```

Comment:
This is actually redundant now.  The cython could be refactored even further.  @astrofrog, could you review the cython changes?